tests/integration/clientv3/cluster_test.go

```diff
@@ -376,31 +377,41 @@ func TestMemberPromoteMemberNotExist(t *testing.T) {
 	}
 }
 
-// TestMaxLearnerInCluster verifies that the maximum number of learners allowed in a cluster is 1
+// TestMaxLearnerInCluster verifies that the maximum number of learners allowed in a cluster
 func TestMaxLearnerInCluster(t *testing.T) {
 	integration2.BeforeTest(t)
 
-	// 1. start with a cluster with 3 voting member and 0 learner member
-	clus := integration2.NewClusterV3(t, &integration2.ClusterConfig{Size: 3})
+	// 1. start with a cluster with 3 voting member and max learner 2
+	clus := integration2.NewClusterV3(t, &integration2.ClusterConfig{Size: 3, ExperimentalMaxLearners: 2})
 	defer clus.Terminate(t)
 
-	// 2. adding a learner member should succeed
-	resp1, err := clus.Client(0).MemberAddAsLearner(context.Background(), []string{"http://127.0.0.1:1234"})
+	// 2. adding 2 learner members should succeed
+	for i := 0; i < 2; i++ {
+		_, err := clus.Client(0).MemberAddAsLearner(context.Background(), []string{fmt.Sprintf("http://127.0.0.1:123%d", i)})
+		if err != nil {
+			t.Fatalf("failed to add learner member %v", err)
+		}
+	}
+
+	// ensure client endpoint is voting member
+	leaderIdx := clus.WaitLeader(t)
+	capi := clus.Client(leaderIdx)
+	resp1, err := capi.MemberList(context.Background())
 	if err != nil {
-		t.Fatalf("failed to add learner member %v", err)
+		t.Fatalf("failed to get member list")
 	}
 	numberOfLearners := 0
 	for _, m := range resp1.Members {
 		if m.IsLearner {
 			numberOfLearners++
 		}
 	}
-	if numberOfLearners != 1 {
-		t.Fatalf("Added 1 learner node to cluster, got %d", numberOfLearners)
+	if numberOfLearners != 2 {
+		t.Fatalf("added 2 learner node to cluster, got %d", numberOfLearners)
 	}
 
-	// 3. cluster has 3 voting member and 1 learner, adding another learner should fail
-	_, err = clus.Client(0).MemberAddAsLearner(context.Background(), []string{"http://127.0.0.1:2345"})
+	// 3. cluster has 3 voting member and 2 learner, adding another learner should fail
+	_, err = clus.Client(0).MemberAddAsLearner(context.Background(), []string{"http://127.0.0.1:2342"})
```

Comment:
Nit: Any reason why we're changing this and the number below on L424?

Reply:
The test was changed to exercise learner counts above the initial default. The port numbers are sequential, 234{0,1,2....} but as a follow-up, I will make this more clear so that no hardcoded values exist.